CONTRIBUTING.md

```diff
@@ -190,15 +190,46 @@ of their patch is not influenced by their setup or implicit knowledge of the sys
 
 ## Code Style
 
-We use automated formatting with Fourmolu to enforce a unified style across the code bases. It is checked in the CI process.
-After installing Fourmolu 0.12, there are some makefile targets to help formatting
-the code base.
-
+We use automated formatting, checked in the CI process, to enforce a unified
+style across the code bases and include these makefile targets to help with
+formatting:
 
 * `make style` - Format the `Cabal`, `Cabal-syntax` and `cabal-install` directories.
 * `make style-modified` - Format files modified in the current tree.
 * `make style-commit COMMIT=<ref>` - Format files modified between HEAD and the given reference.
 
+> [!NOTE]
+> We use `fourmolu-0.12.0.0` for formatting. If installing it with `cabal
+> install`, please make sure to use a version of `GHC >= 9.2.1 && < 9.8`.
+> Its requirement of `GHC2021` sets the lower bound, and its reliance on
+> `ghc-lib-parser` sets the upper bound on GHC versions. A command for
+> installing it this way is:
+>
+> ```
+> $ cabal install fourmolu-0.12.0.0 --overwrite-policy=always --ignore-project
```

Comment:
A different worry here is that people may have other versions of fourmolu installed, and may need them for other projects. Sadly, there is no good solution to this at present.

Reply:
With already having installed `fourmolu` once before, reinstalling is much quicker, isn't it? For me, this seems to be almost instant and, I would guess, only involves resetting the symlink, doesn't it?

Reply:
There's nothing more to add to the contributing docs about this, is there?

Reply:
@geekosaur are you happy to resolve this thread?

Reply:
I'd prefer a warning about having to switch `fourmolu` versions, possibly with a reference to #9230, but otherwise it's fine.

Reply:
Well, except [we already have problems with `--program-suffix`](https://github.com/haskell/cabal/issues/10290) and I really wouldn't be surprised if they also extend to `--program-prefix` as well. (Although, isn't that suffix instead of prefix?)

Reply:
@geekosaur, I added a warning about switching fourmolu versions and referenced #9230.